Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -1020,9 +1020,8 @@ bool RenderFlexibleBox::childCrossSizeShouldUseContainerCrossSize(const RenderBo
         if (crossAxisIsPhysicalWidth())
             return true;
         // This must be kept in sync with computeMainSizeFromAspectRatioUsing().
-        // FIXME: so far we're only considered fixed sizes but we should extend it to other definite sizes.
         auto& crossSize = isHorizontalFlow() ? style().height() : style().width();
-        return crossSize.isFixed();
+        return crossSize.isFixed() || (crossSize.isPercent() && availableLogicalHeightForPercentageComputation()); 
     }
```

Comment:
This could probably be just:
return hasDefiniteLogicalHeight();

I am a bit concerned about the 0px cross size value we are going to return when we fail to resolve the definite value. It could produce a collapsed flex item.

We could do something like this:

```
std::optional<LayoutUnit> RenderFlexibleBox::computeCrossSizeForChildUsingContainerCrossSize(const RenderBox& child) const
{
    if (crossAxisIsPhysicalWidth())
        return contentWidth();

    // Let's compute the definite size value for the flex item (value that we can resolve without running layout).
    auto isHorizontal = isHorizontalFlow();
    auto resolvedValue = [&] () -> std::optional<LayoutUnit> {
        auto size = isHorizontal ? style().height() : style().width();
        if (size.isFixed())
            return { LayoutUnit { size.value() } };
        if (size.isPercent()) {
            auto resolvedPercentValue = availableLogicalHeightForPercentageComputation();
            // childCrossSizeShouldUseContainerCrossSize said we can resolve percent values.
            ASSERT(resolvedPercentValue);
            return resolvedPercentValue;
        }
        ASSERT_NOT_REACHED();
        return { };
    }();

    if (!resolvedValue)
        return { };
    // Keep this sync'ed with childCrossSizeShouldUseContainerCrossSize().
    auto definiteSizeValue = [&] {
        auto definiteValue = *resolvedValue;
        auto maximumSize = isHorizontal ? style().maxHeight() : style().maxWidth();
        if (maximumSize.isFixed())
            definiteValue = std::min(*resolvedValue, LayoutUnit { maximumSize.value() });

        auto minimumSize = isHorizontal ? style().minHeight() : style().minWidth();
        if (minimumSize.isFixed())
            definiteValue = std::max(*resolvedValue, LayoutUnit { minimumSize.value() });

        return definiteValue;
    };
    return { std::max(0_lu, definiteSizeValue() - crossAxisMarginExtentForChild(child)) };
}  
```
However it only helps with one of the 2 callsites:
 _computeChildIntrinsicLogicalWidths()_  -> could fallback to an auto override size scope (same behavior as before)
 _computeMainSizeFromAspectRatioUsing()_ ->  needs a way to return a sane value (the safest would be to propagate the optional value all the way back to _computeFlexBaseSizeForChild()_ -a rather verbose change)

Or, instead of a band-aid, we could address the actual issue and let _childCrossSizeShouldUseContainerCrossSize()_ return the  definite value that we can somehow pass it in to _computeCrossSizeForChildUsingContainerCrossSize()_  (another verbose change). 
It would certainly eliminate the uncertainty introduced by this multi layer structure where the decision point and the place where we try to resolve the value are so disconnected from each other.